hudi-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadata.java

```diff
@@ -309,6 +306,13 @@ public HoodieBackedTableMetadata(Configuration conf, String datasetBasePath, Str
       }
     }
 
+    // Retrieve record from unsynced timeline instants
```

Comment:
Looks like a good amount of code was reusuable. I am wondering if we can do this in a separate `TimelineMergedTableMetadata` implementation, that wraps `HoodieBackedTableMetadata`?